synapse/config/registration.py

```diff
@@ -52,6 +54,23 @@ def default_config(self, **kwargs):
         # Enable registration for new users.
         enable_registration: False
 
+        # The user must provide all of the below types of 3PID when registering.
+        #
+        # registrations_require_3pid:
+        #     - email
+        #     - msisdn
```

Comment:
Personally I would have the two config option entirely separate, as this doesn't support the use case of "require a work email but let people bind personal emails"

Reply:
we're not trying to solve for that use case here, plus having talked to @dbkr it's hard to do anyway - as /account/3pid goes and adds the email to your account (letting you log in with it) as well as optionally binding it on the identity server... so you can't separate the two easily anyway.

Hence me splitting it into:
 * What 3PID media do we mandate at registration?
 * What format 3PIDs are allowed to be added to accounts on this HS?
...with no restrictions on formats valid on ISes.

Reply:
btw, this is a good example of a review comment where I'm staring at it thinking "does he really want me to rewrite it again? or is he just voicing a random comment?".  It might be worth clearly spelling out which comments actually block accepting the PR from your POV, otherwise the committer has no option than assume everything is a request for change...

Reply:
Ok, in this case it was an observation that this feels odd and questioning whether this is the right thing to do. The synapse review process is not just about whether functional changes are right, but if we should even be doing it like that.